controller/apps_authclient.go

```diff
@@ -0,0 +1,164 @@
+package controller
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	rest "k8s.io/client-go/rest"
+)
+
+// AuthClient contains configuration and methods for interacting with OSIO API
+type AuthClient struct {
+	config *rest.Config
+}
+
+// AuthToken is an authorization token returned from the OSIO Auth API
+// currently, github and openshift
+type AuthToken struct {
+	AccessToken *string `json:"access_token,omitempty"`
+	Scope       *string `json:"scope,omitempty"`
+	TokenType   *string `json:"token_type,omitempty"`
+}
+
+// AuthUser is a user record returned from the OSIO Auth API
+// this is only a subset of the record
+type AuthUser struct {
+	Data struct {
+		Attributes struct {
+			Cluster  *string
+			UserID   *string
+			Username *string
+		}
+	}
+}
+
+// NewAuthClient creates an openshift IO client given an http request context
+func NewAuthClient(authToken string, authURL string) *AuthClient {
+
+	config := rest.Config{
+		Host:        authURL,
+		BearerToken: authToken,
+	}
+
+	client := new(AuthClient)
+	client.config = &config
+
+	return client
+}
+
+func (authclient *AuthClient) getAuthToken(forHost string) (*AuthToken, error) {
+	var body []byte
+
+	fullURL := strings.TrimSuffix(authclient.config.Host, "/") + "/api/token?for=" + forHost
```

Comment:
There is a better way to call Auth service: https://github.com/fabric8-services/fabric8-wit/blob/master/auth/resource.go#L63
No need to hardcode paths. Create the generated auth client instead. It will also forward needed headers to the request context for you (authorization and request ID).

Reply:
With your PR pending, I've deleted apps_authclient and recoded using the generated auth client instead.